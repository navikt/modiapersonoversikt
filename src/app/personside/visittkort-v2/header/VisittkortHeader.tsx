import * as React from 'react';
import { Data as Persondata, Kjonn, KodeBeskrivelse } from '../PersondataDomain';
import { Undertittel } from 'nav-frontend-typografi';
import styled from 'styled-components/macro';
import theme, { pxToRem } from '../../../../styles/personOversiktTheme';
import { hentAlderEllerDod, hentNavn } from '../visittkort-utils';
import { useRef } from 'react';
import PersonStatus from './status/PersonStatus';
import Etiketter from './etiketter/Etiketter';
import VisMerChevron from '../../../../components/VisMerChevron';
import NavKontorContainer from './navkontor/NavKontorContainer';
import { useOnMount } from '../../../../utils/customHooks';
import KjonnIkon from './KjonnIkon';

interface Props {
    persondata: Persondata;
    erApen: boolean;
    toggleApen: () => void;
}

const VisittkortHeaderDiv = styled.section`
    background-color: white;
    padding: ${theme.margin.layout};
    padding-right: 0;
    position: relative;
    display: flex;
    flex-wrap: nowrap;
    width: 100%;
    cursor: pointer;
`;

const StyledContent = styled.div`
    flex-grow: 1;
    display: flex;
    flex-flow: row wrap;
    > * {
        flex: 1 1 15rem;
    }
`;

const VenstreFelt = styled.section`
    display: flex;
    min-width: 35%;
`;

const HoyreFelt = styled.div`
    display: flex;
    flex-flow: column nowrap;
    justify-content: space-between;
    text-align: right;
    box-sizing: border-box;
`;

const GrunninfoDiv = styled.section`
    flex: 1 1;
    text-align: left;
    display: flex;
    flex-flow: column nowrap;
    justify-content: space-between;
    word-break: break-word;
    *:focus {
        ${theme.focus}
    }
    > *:first-child {
        margin-bottom: 0.2em !important;
    }
`;

const ChevronStyling = styled.div`
    padding: ${pxToRem(10)};
    display: flex;
    align-items: center;
    justify-content: center;
`;

const ukjentKjonn: KodeBeskrivelse<Kjonn> = {
    kode: Kjonn.U,
    beskrivelse: 'Ukjent kjønn'
};

function VisittkortHeader(props: Props) {
    const navneLinjeRef = useRef<HTMLSpanElement>(null);
    const person = props.persondata.person;

    useOnMount(() => {
        if (!person.sikkerhetstiltak.isEmpty()) {
            return;
        }
        navneLinjeRef.current?.focus();
    });

    const kjonn = person.kjonn.firstOrNull() ?? ukjentKjonn;
    const navn = person.navn.firstOrNull();

    return (
        <VisittkortHeaderDiv role="region" aria-label="Visittkort-hode" onClick={props.toggleApen}>
            <StyledContent>
                <VenstreFelt>
                    <KjonnIkon kjonn={kjonn.kode} />
                    <GrunninfoDiv>
                        <Undertittel tag="h1">
                            <span ref={navneLinjeRef} tabIndex={-1} /* for at focus skal funke */>
                                {hentNavn(navn)} ({hentAlderEllerDod(person)})
                            </span>
                        </Undertittel>
                        <span className="visually-hidden">{kjonn.beskrivelse}</span>
                        <PersonStatus person={person} />
                    </GrunninfoDiv>
                </VenstreFelt>
                <HoyreFelt>
                    <Etiketter person={person} />
                    <NavKontorContainer person={person} />
                </HoyreFelt>
            </StyledContent>
            <ChevronStyling>
                <VisMerChevron
                    onClick={(e) => {
                        e.stopPropagation();
                        props.toggleApen();
                    }}
                    open={props.erApen}
                    title={(props.erApen ? 'Lukk' : 'Åpne') + ' visittkort (Alt + N)'}
                    focusOnRelativeParent={true}
                >
                    <span className="visually-hidden">{props.erApen ? 'Lukk visittkort' : 'Ekspander visittkort'}</span>
                </VisMerChevron>
            </ChevronStyling>
        </VisittkortHeaderDiv>
    );
}

export default VisittkortHeader;
