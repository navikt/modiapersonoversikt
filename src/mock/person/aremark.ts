import { Diskresjonskoder, TilrettelagtKommunikasjonsTyper } from '../../konstanter';
import { Kjønn, Person, Relasjonstype, SivilstandTyper } from '../../models/person/person';
import { SikkerhetstiltakTyper } from '../../models/sikkerhetstiltak';

export const aremark: Person = {
    fødselsnummer: '10108000398',
    kjønn: Kjønn.Mann,
    geografiskTilknytning: '0118',
    alder: 42,
    bankkonto: {
        banknavn: 'Bien sparebank',
        kontonummer: '12345678910',
        sistEndret: '2005-12-12',
        sistEndretAv: 'Daniel'
    },
    sikkerhetstiltak: {
        sikkerhetstiltaksbeskrivelse: 'Får verken møte opp på NAV-kontor eller ringe på telefon ' +
        'grunnet truende oppførsel og språk.',
        sikkerhetstiltakskode: SikkerhetstiltakTyper.FysiskOgTelefonUtestengelse,
        periode: {
            fra: '2005-12-12',
            til: '2025-12-12'
        }
    },
    navn: {
        sammensatt: 'TESTFAMILIEN AREMARK',
        fornavn: 'AREMARK',
        mellomnavn: '',
        etternavn: 'TESTFAMILIEN',
    },
    tilrettelagtKomunikasjonsListe: [
        {
            kodeRef: 'TOHJ',
            beskrivelse: TilrettelagtKommunikasjonsTyper.TOHJ
        }
    ],
    diskresjonskode: { kodeRef: Diskresjonskoder.FORTROLIG_ADRESSE, beskrivelse: 'Sperret adresse, fortrolig' },
    statsborgerskap: { kodeRef: 'NOR', beskrivelse: 'NORGE' },
    personstatus: {
        dødsdato: undefined,
        bostatus: undefined
    },
    sivilstand: {
        kodeRef: SivilstandTyper.Gift,
        beskrivelse: 'Gift',
        fraOgMed: '2005-12-12'
    },
    familierelasjoner: [{
        harSammeBosted: false,
        rolle: Relasjonstype.Barn,
        tilPerson: {
            navn: {
                sammensatt: '',
                fornavn: 'Siri',
                mellomnavn: null,
                etternavn: 'Aremark'
            },
            alder: 17,
            alderMåneder: 184,
            fødselsnummer: '10117100000',
            personstatus: {}
        }

    }],
    kontaktinformasjon: {
        mobil: {
            sistEndret: '2014-06-21T18:44:39+02:00',
            identifikator: '99887766',
            retningsnummer: { kodeRef: '+47', beskrivelse: 'Norge' },
            sistEndretAv: 'BRUKER'
        }
    }
};