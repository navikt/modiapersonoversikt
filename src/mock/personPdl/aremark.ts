import {
    Adresse,
    AdresseBeskyttelse,
    EgenAnsatt,
    Enhet,
    Kjonn,
    KodeBeskrivelse,
    Navn,
    Person,
    PersonStatus,
    Sikkerhetstiltak,
    SikkerhetstiltakType,
    Sivilstand,
    SivilstandType,
    Statsborgerskap,
    Telefon,
    TilrettelagtKommunikasjon
} from '../../models/personPdl/person';
import { getAremarkForeldreansvar } from './foreldreansvarMock';
import { getDeltBostedMock } from './deltBostedMock';
import * as faker from 'faker';
import { getMockDodsbo } from './dodsbo';
import navfaker from 'nav-faker/dist/index';
import { mockVergemal } from './vergemalMock';
import { DigitalKontaktinformasjon } from '../../models/personPdl/digitalKontaktinformasjon';
import { getBankKonto } from './bankkontoMock';
import { Fullmakt, FullmaktsRolle } from '../../models/personPdl/fullmakt';

export const navnAremark: Navn = {
    fornavn: 'TESTFAMILIEN',
    mellomnavn: null,
    etternavn: 'AREMARK'
};

export const kodeBeskrivelseKjonn: KodeBeskrivelse<Kjonn> = {
    kode: Kjonn.Mann,
    beskrivelse: 'Mann'
};

export const adresseAremark: Adresse = {
    linje1: 'Adresseveien 1, 0661 AREMARKSTAD',
    linje2: null,
    linje3: null
};

export const navEnhetAremark: Enhet = {
    id: '0000',
    navn: 'Nav-kontor'
};

export const statsborgerskapAremark: Statsborgerskap = {
    land: {
        kode: 'NOR',
        beskrivelse: 'Norsk statsborger'
    },
    gyldigFraOgMed: new Date('1980-10-10'),
    gyldigTilOgMed: null
};

export const adressebeskyttelseAremark: KodeBeskrivelse<AdresseBeskyttelse> = {
    kode: AdresseBeskyttelse.Ugradert,
    beskrivelse: 'Ingen'
};

export const sikkerhetstiltakAremark: Sikkerhetstiltak = {
    type: SikkerhetstiltakType.TelefoniskUtestengelse,
    gyldigFraOgMed: new Date('2005-02-13'),
    gyldigTilOgMed: new Date('2020-02-02')
};

export const personstatusAremark: KodeBeskrivelse<PersonStatus> = {
    kode: PersonStatus.Bosatt,
    beskrivelse: 'Bosatt i Norge'
};

export const sivilstandAremark: Sivilstand = {
    type: {
        kode: SivilstandType.Skilt,
        beskrivelse: 'Skilt'
    },
    gyldigFraOgMed: new Date('2010-03-06')
};

export const tilrettelagtKommunikasjonAremark: TilrettelagtKommunikasjon = {
    talesprak: [
        {
            kode: 'NO',
            beskrivelse: 'Norsk'
        }
    ],
    tegnsprak: [
        {
            kode: 'NO',
            beskrivelse: 'Norsk'
        }
    ]
};

export const telefonAremark: Telefon = {
    retningsnummer: {
        kode: '+47',
        beskrivelse: 'retningsnummer er norsk'
    },
    identifikator: '99009900',
    sistEndret: new Date('2017-10-10'),
    sistEndretAv: 'TESTFAMILIEN AREMARK',
    prioritet: 1
};

export const digitalKontaktInformasjonAremark: DigitalKontaktinformasjon = {
    personident: '10108000398',
    reservasjon: 'reservert',
    epostadresse: {
        value: null,
        sistOppdatert: null,
        sisVerifisert: new Date('2013-01-01')
    },
    mobiltelefonnummer: {
        value: null,
        sistOppdatert: null,
        sisVerifisert: null
    }
};

export const fullmaktAremark: Fullmakt = {
    motpartsPersonident: '123456789',
    motpartsPersonNavn: {
        fornavn: 'Navn',
        mellomnavn: null,
        etternavn: 'Navnesen'
    },
    motpartsRolle: FullmaktsRolle.Fullmektig,
    omraade: ['*'],
    gyldigFraOgMed: new Date('2019-01-01'),
    gyldigTilOgMed: new Date('2022-12-12')
};

export const aremark: Person = {
    fnr: '10108000398',
    navn: [navnAremark],
    kjonn: [kodeBeskrivelseKjonn],
    fodselsdato: [new Date('1980-10-10')],
    dodsdato: [],
    bostedAdresse: [adresseAremark],
    kontaktAdresse: [adresseAremark],
    navEnhet: navEnhetAremark,
    statsborgerskap: [statsborgerskapAremark],
    adressebeskyttelse: [adressebeskyttelseAremark],
    sikkerhetstiltak: [sikkerhetstiltakAremark],
    erEgenAnsatt: EgenAnsatt.Nei,
    personstatus: [personstatusAremark],
    sivilstand: [sivilstandAremark],
    foreldreansvar: [getAremarkForeldreansvar()],
    deltBosted: getDeltBostedMock(faker),
    dodsbo: getMockDodsbo(faker),
    fullmakt: [fullmaktAremark],
    vergemal: mockVergemal(navfaker.personIdentifikator.f√∏dselsnummer()),
    tilrettelagtKommunikasjon: tilrettelagtKommunikasjonAremark,
    telefonnummer: [telefonAremark],
    kontaktOgReservasjon: digitalKontaktInformasjonAremark,
    bankkonto: getBankKonto()
};
