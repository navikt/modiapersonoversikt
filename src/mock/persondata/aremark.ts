import {
    AdresseBeskyttelse,
    EgenAnsatt,
    ForelderBarnRelasjon,
    ForelderBarnRelasjonRolle,
    FullmaktsRolle,
    Kjonn,
    LocalDate,
    LocalDateTime,
    Person,
    PersonStatus,
    SivilstandType,
    Skifteform
} from '../../app/personside/visittkort-v2/PersondataDomain';
import { harDiskresjonskode } from '../../app/personside/visittkort-v2/visittkort-utils';

export function hentBarnAremark(): ForelderBarnRelasjon[] {
    return [
        {
            ident: '12345678910',
            rolle: ForelderBarnRelasjonRolle.BARN,
            navn: [
                {
                    fornavn: 'BARN1',
                    mellomnavn: null,
                    etternavn: 'BARNESEN'
                }
            ],
            fodselsdato: ['2008-03-15' as LocalDate],
            alder: 13,
            kjonn: [
                {
                    kode: Kjonn.K,
                    beskrivelse: 'Kvinne'
                }
            ],
            adressebeskyttelse: [
                {
                    kode: AdresseBeskyttelse.UGRADERT,
                    beskrivelse: 'UGRADERT'
                }
            ],
            harSammeAdresse: false,
            dodsdato: []
        },
        {
            ident: '12345678911',
            rolle: ForelderBarnRelasjonRolle.BARN,
            navn: [
                {
                    fornavn: 'BARN1',
                    mellomnavn: null,
                    etternavn: 'BARNESEN'
                }
            ],
            fodselsdato: ['2008-03-15' as LocalDate],
            alder: null,
            kjonn: [
                {
                    kode: Kjonn.K,
                    beskrivelse: 'Kvinne'
                }
            ],
            adressebeskyttelse: [],
            harSammeAdresse: false,
            dodsdato: []
        }
    ];
}

export const aremark: Person = {
    fnr: '10108000398',
    navn: [
        {
            fornavn: 'AREMARK',
            mellomnavn: null,
            etternavn: 'TESTFAMILIEN'
        }
    ],
    kjonn: [
        {
            kode: Kjonn.M,
            beskrivelse: 'Mann'
        }
    ],
    fodselsdato: ['1980-10-10' as LocalDate],
    geografiskTilknytning: '0118',
    alder: 41,
    dodsdato: [],
    bostedAdresse: [
        {
            linje1: 'Islandsgate 49',
            linje2: '7693 Hovedøya',
            linje3: 'Norge',
            sistEndret: {
                ident: 'Folkeregisteret',
                tidspunkt: '2010-02-20T10:15:30' as LocalDateTime,
                system: 'FREG'
            },
            gyldighetsPeriode: null
        }
    ],
    kontaktAdresse: [
        {
            linje1: 'Aremarkveien 1',
            linje2: '0320 Aremark',
            linje3: null,
            sistEndret: {
                ident: 'Folkeregisteret',
                tidspunkt: '2020-01-01T10:15:30' as LocalDateTime,
                system: 'FREG'
            },
            gyldighetsPeriode: null
        }
    ],
    oppholdsAdresse: [
        {
            linje1: 'Oppholdsadresse 1',
            linje2: '0000 AREMARK',
            linje3: null,
            sistEndret: null,
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    navEnhet: {
        id: '0219',
        navn: 'NAV Bærum',
        publikumsmottak: [
            {
                besoksadresse: {
                    linje1: 'Adressevei 1',
                    linje2: '0000 AREMARK',
                    linje3: null,
                    sistEndret: null,
                    gyldighetsPeriode: null
                },
                apningstider: [
                    {
                        ukedag: 'Mandag',
                        apningstid: '09.00 - 15.00'
                    },
                    {
                        ukedag: 'Tirsdag',
                        apningstid: '09.00 - 15.00'
                    },
                    {
                        ukedag: 'Onsdag',
                        apningstid: '10.00 - 16.00'
                    }
                ]
            }
        ]
    },
    statsborgerskap: [
        {
            land: {
                kode: 'NOR',
                beskrivelse: 'NORGE'
            },
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    adressebeskyttelse: [
        {
            kode: AdresseBeskyttelse.KODE6,
            beskrivelse: 'Kode 6'
        }
    ],
    sikkerhetstiltak: [
        {
            type: 'TFUS',
            beskrivelse: 'Telefonisk utestengelse',
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    erEgenAnsatt: EgenAnsatt.JA,
    personstatus: [
        {
            kode: PersonStatus.BOSATT,
            beskrivelse: 'Bosatt i Norge'
        }
    ],
    sivilstand: [
        {
            type: {
                kode: SivilstandType.GIFT,
                beskrivelse: 'Gift'
            },
            gyldigFraOgMed: '2005-12-12' as LocalDate,
            sivilstandRelasjon: {
                fnr: '12345555555',
                navn: [
                    {
                        fornavn: 'BAREMARK',
                        mellomnavn: null,
                        etternavn: 'TESTFAMILIEN'
                    }
                ],
                alder: 40,
                adressebeskyttelse: [
                    {
                        kode: AdresseBeskyttelse.UGRADERT,
                        beskrivelse: 'UGRADERT'
                    }
                ],
                harSammeAdresse: true
            }
        }
    ],
    foreldreansvar: hentBarnAremark().map((barn) => ({
        ansvar: 'felles',
        ansvarlig: null,
        ansvarsubject: {
            navn: harDiskresjonskode(barn.adressebeskyttelse) ? null : barn.navn.firstOrNull(),
            ident: barn.ident
        }
    })),
    deltBosted: [
        {
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            },
            adresse: {
                linje1: `Adresseveien 1`,
                linje2: '0000 Aremark',
                linje3: null,
                sistEndret: null,
                gyldighetsPeriode: null
            }
        }
    ],
    dodsbo: [
        {
            adressat: {
                advokatSomAdressat: {
                    kontaktperson: {
                        fornavn: 'Advokat',
                        mellomnavn: null,
                        etternavn: 'Navnesen'
                    },
                    organisasjonsnavn: 'Advokatkontoret Aremark',
                    organisasjonsnummer: '01234567891011'
                },
                organisasjonSomAdressat: null,
                personSomAdressat: null
            },
            adresse: {
                linje1: 'Elgelia 20, 0000 Aremark',
                linje2: null,
                linje3: null,
                sistEndret: null,
                gyldighetsPeriode: null
            },
            registrert: '2010-02-02' as LocalDate,
            skifteform: Skifteform.OFFENTLIG,
            sistEndret: {
                ident: 'Folkeregisteret',
                tidspunkt: '2020-01-01T10:15:30' as LocalDateTime,
                system: 'Folkeregisteret'
            }
        }
    ],
    fullmakt: [
        {
            motpartsPersonident: '123456789',
            motpartsPersonNavn: {
                fornavn: 'Navn',
                mellomnavn: null,
                etternavn: 'Navnesen'
            },
            motpartsRolle: FullmaktsRolle.FULLMEKTIG,
            omrade: [
                {
                    kode: '*',
                    beskrivelse: ''
                }
            ],
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-06-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    vergemal: [
        {
            ident: '21042900076',
            navn: {
                fornavn: 'Simen',
                mellomnavn: null,
                etternavn: 'Solli'
            },
            vergesakstype: 'Voksen',
            omfang: 'Ivareta personens interesser innenfor det personlige og økonomiske området',
            embete: 'Fylkesmannen i Troms og Finnmark',
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-06-01' as LocalDate,
                gyldigTilOgMed: '2022-05-10' as LocalDate
            }
        },
        {
            ident: '123456799',
            navn: {
                fornavn: 'Truls',
                mellomnavn: null,
                etternavn: 'Tøffel'
            },
            vergesakstype: 'Fremtidsfullmakt',
            omfang: 'Ivareta personens interesser innenfor det økonomiske området',
            embete: 'Fylkesmannen i Troms og Finnmark',
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-10' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    tilrettelagtKommunikasjon: {
        talesprak: [
            {
                kode: 'NO',
                beskrivelse: 'Norsk'
            }
        ],
        tegnsprak: [
            {
                kode: 'NO',
                beskrivelse: 'Norsk'
            }
        ]
    },
    telefonnummer: [
        {
            retningsnummer: {
                kode: '+47',
                beskrivelse: 'Norge'
            },
            identifikator: '99009900',
            sistEndret: {
                ident: '11223344',
                tidspunkt: '2018-06-14T00:00:00' as LocalDateTime,
                system: 'BD06'
            },
            prioritet: -1
        }
    ],
    kontaktOgReservasjon: {
        personident: '10108000398',
        reservasjon: 'true',
        epostadresse: {
            value: 'epost@nav.no',
            sistOppdatert: null,
            sistVerifisert: '2013-01-01' as LocalDate
        },
        mobiltelefonnummer: {
            value: '12345678',
            sistOppdatert: '2021-11-02' as LocalDate,
            sistVerifisert: '2021-11-02' as LocalDate
        }
    },
    bankkonto: {
        kontonummer: '12345678900',
        banknavn: 'DNB ASA',
        sistEndret: {
            ident: '1010800 BD03',
            tidspunkt: '2006-03-15T00:00:00' as LocalDateTime,
            system: ''
        },
        bankkode: null,
        swift: 'DNBANOKKXXX',
        landkode: null,
        adresse: {
            linje1: 'Bankveien 1,',
            linje2: '0357 Bankestad',
            linje3: null,
            sistEndret: null,
            gyldighetsPeriode: null
        },
        valuta: {
            kode: 'NOK',
            beskrivelse: 'Norske kroner'
        }
    },
    forelderBarnRelasjon: hentBarnAremark()
};
