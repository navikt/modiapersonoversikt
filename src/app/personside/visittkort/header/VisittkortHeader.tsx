import * as React from 'react';
import styled from 'styled-components';
import Undertittel from 'nav-frontend-typografi/lib/undertittel';
import NavKontorContainer from './navkontor/NavKontorContainer';
import { erDød, Navn, Person } from '../../../../models/person/person';
import PersonStatus from './status/PersonStatus';
import EtiketterContainer from './Etiketter/EtiketterContainer';
import Mann from '../../../../svg/Mann.js';
import Kvinne from '../../../../svg/Kvinne.js';
import VisMerChevron from '../../../../components/VisMerChevron';
import theme from '../../../../styles/personOversiktTheme';
import ToolTip from '../../../../components/tooltip/ToolTip';
import IfFeatureToggleOn from '../../../../components/featureToggle/IfFeatureToggleOn';
import { FeatureToggles } from '../../../../components/featureToggle/toggleIDs';

interface Props {
    visittkortApent: boolean;
    person: Person;
    toggleVisittkort: (erApen?: boolean) => void;
}

interface State {
    showTooltip: boolean;
}

const VisittkortHeaderDiv = styled.section`
    ${theme.hvittPanel};
    padding: ${theme.margin.px20};
    padding-right: 3rem;
    margin-bottom: 0.2rem;
    position: relative;
    width: 100%;
    display: flex;
    flex-flow: row wrap;
    cursor: pointer;
    > * {
        flex: 1 1 50%;
    }
`;

const VenstreFelt = styled.section`
    display: flex;
`;

const HøyreFelt = styled.div`
    display: flex;
    flex-flow: column nowrap;
    justify-content: space-between;
    text-align: right;
    box-sizing: border-box;
`;

const IkonDiv = styled.div`
    flex: 0 0 ${theme.margin.px50};
    text-align: left;
    svg {
        height: ${theme.margin.px40};
        width: auto;
    }
`;

const GrunninfoDiv = styled.section`
    flex: 1 1;
    text-align: left;
    display: flex;
    flex-flow: column nowrap;
    justify-content: space-between;
    *:focus {
        ${theme.focus}
    }
    > *:first-child {
        margin-bottom: 0.2em !important;
    }
`;

const ChevronStyling = styled.div`
    position: absolute;
    right: 1rem;
    top: 50%;
    transform: translateY(-50%);
`;

function hentNavn(navn: Navn) {
    return navn.fornavn + (navn.mellomnavn ? ' ' + navn.mellomnavn + ' ' : ' ') + navn.etternavn;
}

class VisittkortHeader extends React.PureComponent<Props, State> {
    private navneLinjeRef = React.createRef<HTMLSpanElement>();

    constructor(props: Props) {
        super(props);
        this.state = { showTooltip: false };
        this.toggleVisittkort = this.toggleVisittkort.bind(this);
    }

    componentDidMount() {
        if (this.navneLinjeRef.current && !this.props.person.sikkerhetstiltak) {
            this.navneLinjeRef.current.focus();
        }
    }

    toggleVisittkort() {
        this.setState({ showTooltip: true });
        this.props.toggleVisittkort(!this.props.visittkortApent);
    }

    render() {
        const { person, visittkortApent } = this.props;
        const ikon = {
            ikon: person.kjønn === 'M' ? <Mann /> : <Kvinne />
        };
        const alder = erDød(person.personstatus) ? 'Død' : person.alder;
        const kjønn = person.kjønn === 'M' ? 'Mann' : 'Kvinne';
        return (
            <VisittkortHeaderDiv role="region" aria-label="Visittkort-hode" onClick={this.toggleVisittkort}>
                <VenstreFelt>
                    <IkonDiv>{ikon.ikon}</IkonDiv>
                    <GrunninfoDiv>
                        <Undertittel tag="h1">
                            <span ref={this.navneLinjeRef} tabIndex={-1} /* for at focus skal funke*/>
                                {hentNavn(person.navn)} ({alder})<span className="visually-hidden">{kjønn}</span>
                            </span>
                        </Undertittel>
                        <PersonStatus person={person} />
                    </GrunninfoDiv>
                </VenstreFelt>

                <HøyreFelt>
                    <EtiketterContainer />
                    <NavKontorContainer person={person} />
                </HøyreFelt>

                <ChevronStyling>
                    <VisMerChevron
                        onClick={this.toggleVisittkort}
                        open={visittkortApent}
                        title={(visittkortApent ? 'Lukk' : 'Åpne') + ' visittkort (Alt + N)'}
                    >
                        <span className="visually-hidden">
                            {visittkortApent ? 'Lukk visittkort' : 'Ekspander visittkort'}
                        </span>
                    </VisMerChevron>
                </ChevronStyling>
                <IfFeatureToggleOn toggleID={FeatureToggles.Tooltip}>
                    {this.state.showTooltip && <ToolTip>Hurtigtast åpne/lukke visittkort: Alt + N</ToolTip>}
                </IfFeatureToggleOn>
            </VisittkortHeaderDiv>
        );
    }
}

export default VisittkortHeader;
