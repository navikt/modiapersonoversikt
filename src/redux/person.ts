import { Person } from '../models/person';
import { Action, Dispatch } from 'redux';
import { FetchError, FetchSuccess, RestActions, STATUS } from './utils';
import { getPerson } from '../api/person-api';
import { aremark } from '../mock/person-mock';

export interface PersonState {
    data: Person;
    status: STATUS;
    error: string | null;
}

const reducerName = 'person/';

const actionNames = {
    PENDING: reducerName + STATUS.PENDING,
    OK: reducerName + STATUS.OK,
    ERROR: reducerName + STATUS.ERROR
};

const initialState: PersonState = {
    data: aremark,
    status: STATUS.NOT_STARTED,
    error: null
};

export function hentPerson(fodselsnummer: string) {
    return (dispatch: Dispatch<Action>) => {
        dispatch({ type: actionNames.PENDING});
        return getPerson(fodselsnummer)
            .then((person) => {
                dispatch({type: actionNames.OK, data: person});
                return person;
            })
            .catch((err) => {
                dispatch({type: actionNames.ERROR, error: err});
            });
    };
}

export default function reducer(state: PersonState = initialState, action: RestActions<Person>): PersonState {
    switch (action.type) {
        case actionNames.PENDING:
            return {...state, status: STATUS.PENDING };
        case actionNames.OK:
            return {...state, status: STATUS.OK, data: (<FetchSuccess<Person>> action).data};
        case actionNames.ERROR:
            return {...state, status: STATUS.ERROR, error: (<FetchError> action).error};
        default:
            return state;
    }
}