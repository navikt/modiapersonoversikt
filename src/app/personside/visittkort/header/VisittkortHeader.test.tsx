import * as React from 'react';
import * as renderer from 'react-test-renderer';
import VisittkortHeader from './VisittkortHeader';
import { aremark } from '../../../../mock/person/aremark';
import TestProvider from '../../../../test/Testprovider';
import { mount } from 'enzyme';
import { Person } from '../../../../models/person/person';
import { personinformasjonActionNames } from '../../../../redux/restReducers/personinformasjon';
import { getTestStore } from '../../../../setupTests';

test('viser info om bruker i visittkort-header', () => {
    const visittkortheader = renderer.create(
        <TestProvider>
            <VisittkortHeader person={aremark} visittkortApent={false} toggleVisittkort={() => null} />
        </TestProvider>
    );

    const json = visittkortheader.toJSON();
    expect(json).toMatchSnapshot();
});

test('setter fokus på brukerens navn on mount', () => {
    const person = {
        ...aremark,
        sikkerhetstiltak: undefined
    };
    let customStore = getTestStore();
    customStore.dispatch({ type: personinformasjonActionNames.FINISHED, data: person });
    mount(
        <TestProvider customStore={customStore}>
            <VisittkortHeader person={person} toggleVisittkort={() => null} visittkortApent={false} />
        </TestProvider>
    );

    const focusedElement = document.activeElement;

    if (focusedElement) {
        expect(focusedElement.innerHTML.toLowerCase()).toContain(aremark.navn.sammensatt.toLowerCase());
    } else {
        fail('Ingen activeElement på dokumentet');
    }
});

test('setter fokus på sikkerhetstiltak on mount', () => {
    const person: Person = {
        ...aremark,
        sikkerhetstiltak: {
            sikkerhetstiltaksbeskrivelse: 'Brukeren er farlig',
            sikkerhetstiltakskode: '42',
            periode: {
                fra: 'i dag',
                til: 'i går'
            }
        }
    };
    let customStore = getTestStore();
    customStore.dispatch({ type: personinformasjonActionNames.FINISHED, data: person });
    mount(
        <TestProvider customStore={customStore}>
            <VisittkortHeader person={person} toggleVisittkort={() => null} visittkortApent={false} />
        </TestProvider>
    );

    const focusedElement = document.activeElement;

    if (focusedElement) {
        expect(focusedElement.innerHTML).toContain('Sikkerhetstiltak');
    } else {
        fail('Ingen activeElement på dokumentet');
    }
});
