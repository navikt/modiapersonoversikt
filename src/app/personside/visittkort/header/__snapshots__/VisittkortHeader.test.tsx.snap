// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viser info om bruker i visittkort-header 1`] = `
<div
  className="sc-EHOje iEMKnz"
>
  <div
    className="sc-bZQynM kCagj"
  >
    <div
      className="sc-htoDjs cUPGxf"
      title="Mann"
    >
      <svg
        height={64}
        viewBox="0 0 64 64"
        width={64}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 0a32 32 0 1 0 0 64 32 32 0 0 0 0-64zm0 8a3.9 3.9 0 1 1 0 7.7 3.9 3.9 0 0 1 0-7.8zm10 15.5v10.8c0 .9-.7 1.6-1.6 1.6-1 0-1.8-.7-1.8-1.6v-8.9c0-.4-.3-.8-.7-.8-.4 0-.7.3-.7.7v28.5a2.2 2.2 0 1 1-4.5 0v-18a.7.7 0 0 0-1.4 0v18a2.2 2.2 0 0 1-4.4 0V25.4a.7.7 0 0 0-1.5 0V34.3c0 .9-.8 1.6-1.8 1.6s-1.7-.7-1.7-1.6V23.5v-.7c0-3.4 3.6-5.8 7-6H35c3.4.2 7 2.6 7 6v.7z"
          fill="#3385D1"
        />
      </svg>
    </div>
    <div
      className="sc-dnqmqq cOJMoz"
    >
      <h2
        className="typo-undertittel"
      >
        AREMARK TESTFAMILIEN
         (
        42
        )
      </h2>
      <span
        className="typo-undertekst"
      >
        <span
          className="sc-bxivhb dCGcuR"
        >
          <span
            title="Fødselsnummer"
          >
            10108000398
             
          </span>
        </span>
        <ul
          className="sc-htpNat gxIENh"
        >
          <li
            title="Statsborgerskap"
          >
            Norge
          </li>
          <li
            title="Sivilstand"
          >
            Gift
          </li>
          <li
            title="Barn under 21 år"
          >
            1
             barn under 21 år
          </li>
        </ul>
      </span>
    </div>
  </div>
  <div
    className="sc-gzVnrw dcXaNe"
  >
    <div
      className="sc-ifAKCX eHvGAp"
    >
      <div
        className="etikett etikett--advarsel"
      >
        <span
          className="typo-normal"
        >
          Kode 7
        </span>
      </div>
      <div
        className="etikett etikett--advarsel"
      >
        <span
          className="typo-normal"
        >
          Sikkerhetstiltak
        </span>
      </div>
      <div
        className="etikett etikett--fokus"
      >
        <span
          className="typo-normal"
        >
          Vergemål
        </span>
      </div>
      <div
        className="etikett etikett--fokus"
      >
        <span
          className="typo-normal"
        >
          Tolkehjelp
        </span>
      </div>
    </div>
    <span
      className="typo-undertekst"
    >
      <dl
        className="sc-bwzfXH gvaWfu"
      >
        <dt>
          NAV-kontor
        </dt>
        <dd>
          0118
           
          NAV Aremark
        </dd>
      </dl>
    </span>
  </div>
</div>
`;
